test/logger-log.js

```diff
@@ -1112,3 +1112,76 @@ test('Can proxy to an ingestion endpoint through an http proxy', (t) => {
     })
   })
 })
+
+test('sendUserAgent is `true` by default', (t) => {
+  const logger = new Logger(apiKey, createOptions())
+  const line = 'This request contains the user-agent header'
+  t.plan(2)
+  t.on('end', async () => {
+    nock.cleanAll()
+  })
+
+  nock(logger.url)
+    .matchHeader('user-agent', (headerValue) => {
+      return headerValue.includes('@logdna/logger')
+    })
+    .post('/', (body) => {
+      const payload = body.ls[0]
+      t.equal(payload.line, line)
+      return true
+    })
+    .query(() => { return true })
+    .reply(200, 'Ingester response')
+
+  logger.on('send', (obj) => {
+    t.deepEqual(obj, {
+      httpStatus: 200
+    , firstLine: line
+    , lastLine: null
+    , totalLinesSent: 1
+    , totalLinesReady: 0
+    , bufferCount: 0
+    }, 'Got send event')
+  })
+  logger.log(line)
+})
+
+test('sendUserAgent does not send `user-agent` header if `false`', (t) => {
+  const logger = new Logger(apiKey, createOptions({
+    sendUserAgent: false
+  }))
+  const line = 'This request DOES NOT contain the user-agent header'
```

Comment:
NB: might want to clarify the test titles here too, e.g.:

- "sendUserAgent does not send logger client `user-agent` value if `false`"
- "This request contains the default/fallback user-agent header value"

basically the header will _always_ be sent, this setting just controls our customized user-agent value.

Reply:
yep, fair, fixing